models/relations.py

```diff
@@ -1,52 +1,68 @@
 from db import db
 from models.common import CommonModel
 from util.jsonencode import JsonEncodedDict
+from datetime import datetime
 
 class RelationsModel(db.Model, CommonModel):
     __tablename__ = 'relations'
 
     user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
     id = db.Column(db.Integer, primary_key = True)
-    name = db.Column(db.String(255))
-    birth_year = db.Column(db.Integer) #date
+    first = db.Column(db.String(255))
+    last = db.Column(db.String(255))
+    death_year = db.Column(db.DateTime)
     is_deceased = db.Column(db.Boolean)
     gender = db.Column(db.String(255))
     relation = db.Column(db.String(255))
     notes = db.Column(db.String(500))
     is_step = db.Column(db.Boolean)
     is_adopted = db.Column(db.Boolean)
+    birth_date = db.Column(db.DateTime)
+    lives_in = db.Column(db.String(500))
+    nickname = db.Column(db.String(255))
 
 
 
-    def __init__(self, user_id, name, birth_year, is_deceased, gender, relation, notes, is_step, is_adopted):
+    def __init__(self, user_id, first, last, death_year, is_deceased, gender, relation, notes, is_step, is_adopted, birth_date, lives_in, nickname):
         self.user_id =  user_id
-        self.name = name
-        self.birth_year = birth_year
+        self.first = first
+        self.last = last
+        self.death_year = death_year
         self.is_deceased = is_deceased
         self.gender = gender
         self.relation = relation
-        self.notes =notes
+        self.notes = notes
         self.is_step = is_step
         self.is_adopted = is_adopted
+        self.birth_date = birth_date
+        self.lives_in = lives_in 
+        self.nickname = nickname
 
     def json(self):
         return {
-            "id": self.id    #do we need to return id and user_id?
-            "name": self.name
-            "birth_year": self.birth_year
-            "is_deceased": self.is_deceased
-            "gender": self.gender
-            "relation": self.relation
-            "notes": self.notes
-            "is_step": self.is_step
-            "is_adopted": self.is_adopted
-
+            "id": self.id,
+            "first": self.first,
+            "last": self.last,
+            "death_year": self.death_year.strftime("%Y") if self.death_year else "",
+            "is_deceased": self.is_deceased,
+            "gender": self.gender,
+            "relation": self.relation,
+            "notes": self.notes,
+            "is_step": self.is_step,
+            "is_adopted": self.is_adopted,
+            "birth_date": self.birth_date.strftime("%m/%d/%Y") if self.birth_date else "", 
+            "lives_in": self.lives_in, 
+            "nickname": self.nickname
         }
 
 
     @classmethod
     def find_by_user_id(cls, user_id):
         return cls.query.filter_by(user_id = user_id).all()
 
+    @classmethod
```

Comment:
don't need this. This classmethod should be taken from the parent class, Common Model.